components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/factory/UnregistrationRequestFactoryTest.java

```diff
@@ -0,0 +1,114 @@
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityRequestFactory;
+import org.wso2.carbon.identity.oauth.dcr.model.UnregistrationRequest;
+import org.wso2.carbon.identity.testutil.powermock.PowerMockIdentityBaseTest;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.HttpMethod;
+
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.doAnswer;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.support.membermodification.MemberMatcher.methodsDeclaredIn;
+import static org.powermock.api.support.membermodification.MemberModifier.suppress;
+import static org.testng.Assert.assertEquals;
+
+@PrepareForTest(UnregistrationRequestFactory.class)
+public class UnregistrationRequestFactoryTest extends PowerMockIdentityBaseTest{
+
+    @Mock
+    UnregistrationRequest.DCRUnregisterRequestBuilder unregisterRequestBuilder;
+
+    @Mock
+    private HttpServletRequest mockHttpRequest;
+
+    @Mock
+    private HttpServletResponse mockHttpResponse;
+
+    private UnregistrationRequestFactory registrationRequestFactory;
+
+    @BeforeMethod
+    private void setUp() {
+
+        registrationRequestFactory = new UnregistrationRequestFactory();
+    }
+
+    /**
+     * DataProvider: requestURI, httpMethod, expected value
+     */
+    @DataProvider(name = "httpMethodAndUriProvider")
+    public Object[][] getHttpMethodAndUri() {
+
+        return new Object[][]{
+                {"dummyVal/identity/register/dummyVal", HttpMethod.DELETE, true},
+                {"dummyVal/identity/register/", HttpMethod.DELETE, false},
+        };
+    }
+
+    @Test(dataProvider = "httpMethodAndUriProvider")
+    public void testCanHandle(String requestURI, String httpMethod, boolean expected ) throws Exception {
+
+        when(mockHttpRequest.getRequestURI()).thenReturn(requestURI);
+        when(mockHttpRequest.getMethod()).thenReturn(httpMethod);
+        assertEquals(registrationRequestFactory.canHandle(mockHttpRequest, mockHttpResponse), expected,
+                "Redirect Uri doesn't match");
+    }
+
+    @Test
+    public void testCreate() throws Exception {
+
+        String dummyApplicationName = "dummyApplicationName";
+        String dummyUserId = "dummyUserId";
+        String dummyConsumerKey = "dummyConsumerKey";
+
+        suppress(methodsDeclaredIn(HttpIdentityRequestFactory.class));
+        when(mockHttpRequest.getRequestURI()).thenReturn("dummyVal/identity/register/" + dummyConsumerKey);
+        when(mockHttpRequest.getMethod()).thenReturn(HttpMethod.DELETE);
+        when(mockHttpRequest.getParameter("applicationName")).thenReturn(dummyApplicationName);
```

Comment:
Are these used as private constants inside the class which is being tested ?. If these are available as public constants please use them.

Reply:
Inside the class which is being tested doesn't have any variable.